### ISSUE TITLE
error[E0599]: the method `map` exists for reference `&ChunkedArray<Utf8Type>`, but its trait bounds were not satisfied

### ISSUE THREAD

**Issue by aptr322** [pola-rs/polars#1455]
#### Are you using Python or Rust?
rust

#### What version of polars are you using?
current

#### What operating system are you using polars on?
macos and linux

#### Describe your bug.

Compilation error:

error[E0599]: the method `map` exists for reference `&ChunkedArray<Utf8Type>`, but its trait bounds were not satisfied
   --> src/main.rs:98:10
    |
98  |         .map(|v| domain_from_url(v.unwrap()))
    |          ^^^ method cannot be called on `&ChunkedArray<Utf8Type>` due to unsatisfied trait bounds
    | 
   ::: /Users/petrov/.cargo/git/checkouts/polars-a7cd433de180eb5f/5241673/polars/polars-core/src/chunked_array/mod.rs:146:1
    |
146 | pub struct ChunkedArray<T> {
    | -------------------------- doesn't satisfy `ChunkedArray<Utf8Type>: Iterator`
    | 
   ::: /Users/petrov/.cargo/git/checkouts/polars-a7cd433de180eb5f/5241673/polars/polars-core/src/datatypes.rs:20:1
    |
20  | pub struct Utf8Type {}
    | -------------------
    | |
    | doesn't satisfy `<Utf8Type as PolarsPrimitiveType>::Native = _`
    | doesn't satisfy `Utf8Type: PolarsNumericType`
    |
    = note: the following trait bounds were not satisfied:
            `Utf8Type: PolarsNumericType`
            `<Utf8Type as PolarsPrimitiveType>::Native = _`
            `&ChunkedArray<Utf8Type>: Iterator`
            which is required by `&mut &ChunkedArray<Utf8Type>: Iterator`
            `ChunkedArray<Utf8Type>: Iterator`
            which is required by `&mut ChunkedArray<Utf8Type>: Iterator`

#### What are the steps to reproduce the behavior?

```
    let s1 = df.column("referer")?
        .drop_nulls().unique()?;
    println!("{:?}", s1);

    let s2 = s1
        .utf8().unwrap()
        // .into_iter()                                                                                                                       
        .map(|v| domain_from_url(v.unwrap()))
        .collect::<Utf8Chunked>()
        .into_series();
```

**Example**

If we cannot reproduce the bug, it is unlikely that we will be able fix it.

#### What is the actual behavior?

Show the query you ran and the actual output. 

If the output is large, put it in a gist: https://gist.github.com/

If the output is small, put it in code fences:

```
your
output
goes
here
```

#### What is the expected behavior?

What do you think polars should have done?

**Comment by ritchie46** [comment]
>```
    let s1 = df.column("referer")?
        .drop_nulls().unique()?;
    println!("{:?}", s1);

    let s2 = s1
        .utf8().unwrap()
        // .into_iter()                                                                                                                       
        .map(|v| domain_from_url(v.unwrap()))
        .collect::<Utf8Chunked>()
        .into_series();
```
You've commented out `into_iter()`?

**Comment by aptr322** [comment]
intentionally, I was looking for into_par_iter

**Comment by aptr322** [comment]
@ritchie46 So I have a series and need to apply a function to each row in parallel. How do I do that? The series contains strings.

Likely a naive question, my rust is at the very beginning

**Comment by ritchie46** [comment]
The ChunkedArrays do not implement rayons parallel iterator trait. What you could do is split/slice the array in chunks over each thread and then aggregate/rechunk the result.

So you go from `Utf8Chunked` to `Vec<Utf8Chunked>` to `Utf8Chunked`

**Comment by aptr322** [comment]
Ok, thank you @ritchie46 

Is there plan to implement parallel iterator for it?

**Comment by ritchie46** [comment]
> Is there plan to implement parallel iterator for it?

It is not a priority, but I am open for an implementation.

**Comment by ritchie46** [comment]
I will close this as there is not really a bug.
